### DISCUSSION TITLE
Community preview edition of Teams Auto Attendant/Call Queue Orchestrator (TACO) available

### DISCUSSION THREAD

**Discussion by Josverl** [OfficeDev/TACO#5] [Announcements]
Teams Auto Attendant/Call Queue Orchestrator (TACO)

We are happy to announce the community preview edition of the Teams Auto Attendant/Call Queue Orchestrator (TACO) previously called the Teams Auto Attendant and Call Queue Management App. The last few weeks we have been working hard with the V-team to finish the app, making GUI changes to improve the app and automating the deployment as much as possible.

The TACO tool can be found in this [OfficeDev/TACO repo](https://github.com/OfficeDev/TACO#readme)

The documentation is available in the Wiki [Home · OfficeDev/TACO Wiki (github.com)](https://github.com/OfficeDev/TACO/wiki)

The TACO app is released as community tool so everybody who wants to help is more than welcome to contribute and improve the app.

**Comment by peakers1976** [comment]
Hi @Josverl,

I have had a couple of attempts at deploying the application with little success. It could be that I am missing something but I have followed the deployment guide step by step.

The deployment of the script appears to work and I do not get any failure output.

![image](https://user-images.githubusercontent.com/18456459/180723845-054e5c19-4f32-46b7-9fb9-e6b30adbfaed.png)

The first problem I encountered is the secrets list/get permissions do not get applied for the service account, I need to manually add the account, the only accounts that get added are the app service principal name and the admin account I run the script.

![image](https://user-images.githubusercontent.com/18456459/180724355-afda39fa-6645-4ad2-bbbc-b59661ebc5a4.png)

The second problem is when I import the solution into PowerApps. the documentation states that I should be able to click next and define the variables, this does not happen, it only gives me the option to import.

![image](https://user-images.githubusercontent.com/18456459/180724826-0e2faedd-e61b-4a92-9f85-df2762cde060.png)

![image](https://user-images.githubusercontent.com/18456459/180725145-713a634f-723a-47d3-8018-aa2c93fd5ba2.png)

I then have to manually go through and update the variables in the solution 

VAR - FunctionKey: value of the host function key
VAR - Tenant: tenantname, for example contoso
VAR - application id: Application (Client) ID of the App registration in Azure AD
VAR - FunctionApp: name of the function app
VAR - TeamsvoicemanagementSPSite: SharePoint Site name

additionally this variable 

![image](https://user-images.githubusercontent.com/18456459/180725494-59c1fafb-250e-44f4-aa85-5641bd11280b.png)

After I update the SharePoint site, I need to edit all of the Sharepoint list entries 

![image](https://user-images.githubusercontent.com/18456459/180725658-a1215498-3f68-4dd9-8d6b-4b3d6387dbca.png)

Finally, none of the Cloud Flows are enabled by default "Not sure if this is by design" if i use the import-CallQueue flow to import auto attendants as per below steps:

Automatically adding Auto Attendants
Perform the following steps to import the configuration of the call queues into the SharePoint Lists:

Browse to https://make.powerapps.com/
From the left menu select solutions
In the solutions overview click on the ... next to Auto Attendant and Call Queue Management and select the option Edit
Search for the item Import-CallQueue click on the ... and select Details followed by Details in a new tab
From the top menu bar select the option run
Wait until the flow has completed

 first, I notice that the Get Secret has the incorrect secret name.

![image](https://user-images.githubusercontent.com/18456459/180726500-4dcb8e1d-f2f6-4004-919c-15d99f1217c8.png)

 I need to update it with the correct result

![image](https://user-images.githubusercontent.com/18456459/180726641-02d45846-07eb-43e0-bfc5-5d2c38b15133.png)

The URI is incorrect for the POST method so I needed to update that.

Finally, it fails with the following error 

![image](https://user-images.githubusercontent.com/18456459/180727075-6879e31c-6bf3-4ff6-b77e-504523f79be0.png)

Any advice would be appreciated as it seems to be that some automation is not working as it should.

Thank you

**Reply by Josverl** [reply]
@peakers1976 , 
I moved your Q to the Q&A Section here : https://github.com/OfficeDev/TACO/discussions/8

**Reply by JohanVeldhuis** [reply]
@peakers1976 I posted an update on the Q&A section. For the last item it looks like that that we don't get the expected data back. Can you share some more details?
